src/mscorlib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -1138,48 +1169,89 @@ private static TransitionTime TZif_CreateTransitionTimeFromPosixRule(string date
                 }
 
-                DateTime timeOfDay;
-                TimeSpan? timeOffset = TZif_ParseOffsetString(time);
-                if (timeOffset.HasValue)
-                {
-                    // This logic isn't correct and can't be corrected until https://github.com/dotnet/corefx/issues/2618 is fixed.
-                    // Some time zones use time values like, "26", "144", or "-2".
-                    // This allows the week to sometimes be week 4 and sometimes week 5 in the month.
-                    // For now, strip off any 'days' in the offset, and just get the time of day correct
-                    timeOffset = new TimeSpan(timeOffset.Value.Hours, timeOffset.Value.Minutes, timeOffset.Value.Seconds);
-                    if (timeOffset.Value < TimeSpan.Zero)
-                    {
-                        timeOfDay = new DateTime(1, 1, 2, 0, 0, 0);
-                    }
-                    else
-                    {
-                        timeOfDay = new DateTime(1, 1, 1, 0, 0, 0);
-                    }
-
-                    timeOfDay += timeOffset.Value;
-                }
-                else
+                return TransitionTime.CreateFloatingDateRule(ParseTimeOfDay(time), month, week, day);
+            }
+            else
+            {
+                if (date[0] != 'J')
                 {
-                    // default to 2AM.
-                    timeOfDay = new DateTime(1, 1, 1, 2, 0, 0);
+                    // should be n Julian day format which we don't support. 
+                    // 
+                    // This specifies the Julian day, with n between 0 and 365. February 29 is counted in leap years.
+                    //
+                    // n would be a relative number from the begining of the year. which should handle if the 
+                    // the year is a leap year or not.
+                    // 
+                    // In leap year, n would be counted as:
+                    // 
+                    // 0                30 31              59 60              90      335            365
+                    // |-------Jan--------|-------Feb--------|-------Mar--------|....|-------Dec--------|
+                    //
+                    // while in non leap year we'll have 
+                    // 
+                    // 0                30 31              58 59              89      334            364
+                    // |-------Jan--------|-------Feb--------|-------Mar--------|....|-------Dec--------|
+                    //
+                    // 
+                    // For example if n is specified as 60, this means in leap year the rule will start at Mar 1,
+                    // while in non leap year the rule will start at Mar 2.
+                    // 
+                    // If we need to support n format, we'll have to have a floating adjustment rule support this case.
```

Comment:
Should we open an issue for this?  or should we wait until we get more feedback to support this format?

Reply:
I'll open a future issue anyway so we may pro-actively fixing it if we get a chance.